File: businesslib/src/main/java/com/wujia/businesslib/BusApiService.java
package com.wujia.businesslib;


import com.wujia.businesslib.data.ApiResponse;
import com.wujia.businesslib.data.MsgDto;
import com.wujia.businesslib.data.VersionBean;

import java.util.List;

import io.reactivex.Flowable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;


public interface BusApiService {

    @GET("/v1/message/findListByUserId")
    Flowable<ApiResponse<MsgDto>> getMsg(@Query("familyId") String familyId, @Query("type") String type, @Query("status") String status, @Query("pageNum") int pageNo, @Query("pageSize") int pageSize);

    @GET("/v1/message/findTopThreeByUserId")
    Flowable<ApiResponse<List<MsgDto.ContentBean>>> getTop3UnReadMsg(@Query("familyId")String familyId);

    @GET("/v1/message/isUnRead")
    Flowable<ApiResponse<Boolean>> isUnReadMsg();


    @FormUrlEncoded
    @POST("/v1/message/updateIsRead")
    Flowable<ApiResponse<Object>> readMsg(@Field("messageId") String id);

    @FormUrlEncoded
    @POST("/v1/service/subscribe")
    Flowable<ApiResponse<Object>> subscribe(@Field("serviceId") String serviceId, @Field("isSubscribe") String isSubscribe);

    @GET("/v1/system/version")
    Flowable<VersionBean> checkVersion();

    @FormUrlEncoded
    @POST("/system/updateVer")
    Flowable<ApiResponse<Object>> updateVer(@Field("versionName")String versionName,@Field("key") String key,@Field("versionCode") String versionCode);
//    Flowable<ApiResponse<Object>> updateVer(@Body VersionUpdate versionUpdate);
}

File: businesslib/src/main/java/com/wujia/businesslib/BusApiService.kt
package com.wujia.businesslib


import com.wujia.businesslib.data.ApiResponse
import com.wujia.businesslib.data.MsgDto
import com.wujia.businesslib.data.VersionBean

import io.reactivex.Flowable
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query


interface BusApiService {

    @get:GET("/v1/message/isUnRead")
    val isUnReadMsg: Flowable<ApiResponse<Boolean>>

    @GET("/v1/message/findListByUserId")
    fun getMsg(@Query("familyId") familyId: String, @Query("type") type: String, @Query("status") status: String, @Query("pageNum") pageNo: Int, @Query("pageSize") pageSize: Int): Flowable<ApiResponse<MsgDto>>

    @GET("/v1/message/findTopThreeByUserId")
    fun getTop3UnReadMsg(@Query("familyId") familyId: String): Flowable<ApiResponse<List<MsgDto.ContentBean>>>


    @FormUrlEncoded
    @POST("/v1/message/updateIsRead")
    fun readMsg(@Field("messageId") id: String): Flowable<ApiResponse<Any>>

    @FormUrlEncoded
    @POST("/v1/service/subscribe")
    fun subscribe(@Field("serviceId") serviceId: String, @Field("isSubscribe") isSubscribe: String): Flowable<ApiResponse<Any>>

    @GET("/v1/system/version")
    fun checkVersion(): Flowable<VersionBean>

    @FormUrlEncoded
    @POST("/system/updateVer")
    fun updateVer(@Field("versionName") versionName: String, @Field("key") key: String, @Field("versionCode") versionCode: String): Flowable<ApiResponse<Any>>
    //    Flowable<ApiResponse<Object>> updateVer(@Body VersionUpdate versionUpdate);
}
